tests/phpunit/tests/block-template-utils.php

```diff
@@ -161,6 +161,53 @@ public function test_build_block_template_result_from_file() {
 		$this->assertEmpty( $template_part->modified );
 	}
 
+	/**
+	 * @dataProvider data_build_block_template_result_from_file_injects_theme_attribute
+	 */
+	public function test_build_block_template_result_from_file_injects_theme_attribute( $filename, $expected_content ) {
+		$template = _build_block_template_result_from_file(
+			array(
+				'slug' => 'single',
+				'path' => __DIR__ . '/../data/templates/' . $filename,
+			),
+			'wp_template'
+		);
+		$this->assertSame( $expected_content, $template->content );
+	}
+
+	public function data_build_block_template_result_from_file_injects_theme_attribute() {
+		$theme = 'block-theme';
+		return array(
+			array(
+				'template-with-template-part.html',
+				sprintf(
+					'<!-- wp:template-part {"slug":"header","align":"full","tagName":"header","className":"site-header","theme":"%s"} /-->',
+					$theme
+				),
+			),
+			array(
+				'template-with-nested-template-part.html',
+				sprintf(
+					'<!-- wp:group -->
+<!-- wp:template-part {"slug":"header","align":"full","tagName":"header","className":"site-header","theme":"%s"} /-->
+<!-- /wp:group -->',
+					$theme
+				),
+			),
+			array(
+				'template-with-template-part-with-existing-theme-attribute.html',
+				'<!-- wp:template-part {"slug":"header","theme":"fake-theme","align":"full", "tagName":"header","className":"site-header"} /-->',
+			),
+			array(
+				'template.html',
+				'<!-- wp:paragraph -->
+<p>Just a paragraph</p>
+<!-- /wp:paragraph -->',
+			),
+		);
+	}
```

Comment:
Have you run a coverage report yet to verify that all lines/branches have been covered?

Reply:
Out of curiosity, how do I run this type of report?

Even if there isn't a full coverage, I don't think it should be a blocker since the only part of the code that we care about is the following function call that needs to get refactored:

https://github.com/WordPress/wordpress-develop/blob/cc775b9985b1dd9a0c8827df378b21e831b400a1/src/wp-includes/block-template-utils.php#L568

Reply:
Here's a coverage report which includes existing tests and tests included in this PR.

![image](https://github.com/WordPress/wordpress-develop/assets/79332690/1cb0faa5-5923-4089-aab4-274c84b2ec8a)
![image](https://github.com/WordPress/wordpress-develop/assets/79332690/476f3ac1-46a4-4124-902f-4b7b7837ebc6)

Everything that this PR intends to cover appears to be covered. 🙂

In terms of the remaining coverage for this function:
- Line 585 isn't hit by the tests that specifically cover this function.
- Two branches aren't covered.
  - One branch can be covered by testing with a `postTypes` key in `$template_file` and a `$template_type` of `wp_template`.
  - One branch can be covered by ensuring both outcomes of line 572 are met by tests, then testing `$template->title` for the expected result.
- There don't appear to be tests specifically for this function that provide a non-default slug in `$template_file`. This means lines 578-581, while _hit_ by the tests, don't appear to have coverage to ensure that `$template->title` is retained for `wp_template` types with a non-default slug in the `$template_file`.

As the uncovered parts don't seem to be part of this PR's goals, we can follow up to complete coverage for `_build_block_template_result_from_file()`. I'd also suggest that we start to separate the tests for each function into their own files, completing `@ticket`, `@covers`, and any other annotations that need to be added. This makes it easier not to get lost amongst the tests.

Reply:
> Out of curiosity, how do I run this type of report?

For the coverage report above:
- As you might expect, coverage takes a little longer to warm up/run compared to test runs without coverage.
- I use a custom `phpunit.xml` file. [This file](https://gist.github.com/costdev/af35ebea90beb47cd958fb3a8e9c3a30) can be dropped directly into `wordpress-develop` and will be ignored by Git.
  - Note lines 39-45 in that file. Just change that line to the source file for the function under test before running coverage.
  - Without coverage: `phpunit --filter test_build_block_template_result_from_file`.
  - With coverage: `XDEBUG_MODE=coverage phpunit --filter test_build_block_template_result_from_file`.
  - If you use `npm run test:php`, the above command will need to be adjusted for that.
  - Without this `phpunit.xml` file, you'd be calling `XDEBUG_MODE=coverage phpunit --filter test_build_block_template_result_from_file --coverage-html ./tests/phpunit/build/logs/coverage-html --path-coverage`. Ain't nobody got time for that!
  - Since the `--filter` value is written as a partial match on anything with that string in it, this detects the two test methods covering this function:
    - `test_build_block_template_result_from_file()`
    - `test_build_block_template_result_from_file_injects_theme_attribute()`
- Load `http://localhost/wordpress-develop/tests/phpunit/build/logs/coverage-html/` - or your equivalent.
- Click `line` or `branch`, depending on what you're checking, then click the function name.
  - Green = hit by the tests
  - Red = not hit by the tests.
  - Hover over a green line to see which test(s) hit it.
  - Uncovered branches are included further down the page. It's a bit obtuse tbh, so sometimes looking at what's in red, then examining the tests themselves can reveal the uncovered branch more easily. If it's a ternary / multi-condition `if`, I sometimes split these in the source code temporarily to reveal which isn't/aren't being hit.
  - Path coverage needs to run in order to add the `branch` link beside the file. However, while the path coverage results under the `path` link are an indicator of uncovered paths, due to limitations of static analysis, some paths aren't always achievable. Because of this, I'd recommend focusing on line/branch coverage, and consider uncovered paths as a potential indicator that you may need to consider if there's unhappy paths that still need coverage.

Reply:
https://github.com/WordPress/wordpress-develop/pull/5396 should bring us to the full coverage for the function.